pennylane_lightning/core/src/simulators/lightning_gpu/initSV.cu

```diff
@@ -95,6 +106,91 @@ void setStateVector_CUDA_call(GPUDataT *sv, index_type &num_indices,
                                                 indices);
     PL_CUDA_IS_SUCCESS(cudaGetLastError());
 }
+
+/**
+ * @brief The CUDA kernel that multiplies the state vector data on GPU device
+ * by a global phase.
+ *
+ * @param sv Complex data pointer of state vector on device.
+ * @param num_sv Number of state vector elements.
+ * @param phase Complex data pointer of input values (on device).
+ */
+template <class GPUDataT, class index_type>
+__global__ void globalPhaseStateVectorkernel(GPUDataT *sv, index_type num_sv,
+                                             GPUDataT phase) {
+    const unsigned int i = blockIdx.x * blockDim.x + threadIdx.x;
+    if (i < num_sv) {
+        sv[i] = ComplexMul(sv[i], phase);
+    }
```

Comment:
This kind of branching may be inefficient for CUDA kernels. Are we sure the `num_sv` value is at least as wide as a CUDA warp so that each SM is run at full capacity?

Reply:
Given the perceived urgency at the time, I simply mimicked the other L-GPU kernels without giving it much thought. That said, this is a common pattern in the CUDA [doc](https://docs.nvidia.com/cuda/cuda-c-programming-guide/index.html#initialization). I think it is there to prevent out of bound access when the array is too small or when the array length is not divisible by the number of threads per block. So toward the end of the array you can't fill the SMs if the array is not commensurate with the number of SMs, but otherwise the branch should not prevent using the SMs fully.